インクリメンタルサーチ実装

suzukinan opened the pull request:
# WHAT

javascriptsにsearch.jsを作成　編集

user_controllerのindexを作成　編集

usersへindex.json.jbiilderを作成　編集

routes.rb誤字を修正　:indexを追加


# WHY
インクリメンタルサーチを実装してユーザーを探しやすくするため

suzukinan committed:
インクリメンタルサーチ実装

javascriptsにsearch.jsを作成　編集

user_controllerのindexを作成　編集

usersへindex.json.jbiilderを作成　編集

routes.rb誤字を修正　:indexを追加

TOMARUMARU commented on app/controllers/users_controller.rb:
```diff
@@ -1,6 +1,11 @@
 class UsersController < ApplicationController
 
   def index
+    @users = User.where('name LIKE(?)', "%#{params[:keyword]}%")
```
ログインユーザー弾きましょう

suzukinan replied:
修正しました。

suzukinan committed:
インクリメンタルサーチ

app/controllers/users_controller.rb
indexの@user末尾に
.where.not(id: current_user.id)を追加しました。

yuki-uchida commented on app/assets/javascripts/search.js:
```diff
@@ -0,0 +1,71 @@
+
+    var search_list = $("#user-search-result");
+
+    function appendUser(user) {
```
Userの何を追加するのかわかる名前だと良いです。HTMLとか

suzukinan replied:
修正しました

suzukinan committed:
インクリメンタルサーチ

app/assets/javascripts/search.js
６行目
function appendUser(user)から
function appendUserName(user) に変更

41行目
appendUser(user); から
appendUserName(user);へ変更

TOMARUMARU reviewed (approved):
LGTM

suzukinan commented:
ありがとうございます。
